### PR TITLE
Add PermissionService

### PR DESCRIPTION
**Pull request by diversemix**

### CODE REVIEW

**Comment by erezmus** [comment]
The user service is using the authorization header from the context, which means that the concept of an http header into the service. We should be passing the userId instead from the context.

File: src/application/dashboard/resolvers.ts [hunk 1]
```diff
@@ -2,23 +2,28 @@ import { DashboardService } from './service';
...
 import { IResolvers } from 'apollo-server-express';
+import { UserService } from 'src/domain/user';
 
-const resolvers = (dashboard: DashboardService): IResolvers => ({
+const resolvers = (dashboard: DashboardService, userService: UserService): IResolvers => ({
     Query: {
-        async getSubmissions(): Promise<Submission[] | null> {
-            return await dashboard.find();
+        async getSubmissions(_, {}: {}, context): Promise<Submission[] | null> {
+            const user = await userService.getCurrentUser(context.authorizationHeader);
```
Comment (norisalsaadie) [inline src/application/dashboard/resolvers.ts]: Quick observation. This isn't standard with how apollo a decode to userId context. Plus, I noticed this header is passed to a repo.. not sure the repo needs to know what a HTTP header is when we have the user id.

https://www.apollographql.com/docs/apollo-server/security/authentication/

Comment (norisalsaadie) [reply src/application/dashboard/resolvers.ts]: I just noticed Erez made the same comment in a way.

**Comment by diversemix** [comment]
> The user service is using the authorization header from the context, which means that the concept of an http header into the service. We should be passing the userId instead from the context.

@erezmus We cannot do this as the UserId does not give us the user's role. In order to get the user's role we need to query for the user - this involves passing on the authentication to the underlying user management system. (in this case continuum)

File: src/application/dashboard/resolvers.ts (hunk 1 above)
Comment (diversemix) [reply src/application/dashboard/resolvers.ts]: Responded to Erez

**Comment by diversemix** [comment]
So after discussion here .... we are thinking because the construction of the token/header information is in `continuum-adaptor` - maybe it should be the thing that takes the `userId` and then constructs the token again internally?

**Comment by erezmus** [comment]
> > The user service is using the authorization header from the context, which means that the concept of an http header into the service. We should be passing the userId instead from the context.
> 
> @erezmus We cannot do this as the UserId does not give us the user's role. In order to get the user's role we need to query for the user - this involves passing on the authentication to the underlying user management system. (in this case continuum)

Two other solutions: pass the role in the token (as a custom claim), or have the middleware request the user information from continuum-adaptor. I'm preferring the first tbh.

**Comment by diversemix** [comment]
Raised the above ticket to cover off neatening the code in the long term.

Also discussed how we could use Auth0 or similar to improve how auth is done and using key-pairs across services. (probably another ticket).

**Pull request merged**